hermeto/core/scm.py

```diff
@@ -209,3 +240,88 @@ def _reset_git_head(repo: Repo, ref: str) -> None:
             "Failed on checking out the Git repository. Please verify the supplied reference "
             f'of "{ref}" is valid.'
         )
+
+
+class GitSupportConfig:
```

Comment:
Introduction of this class should also be (including the unit test you have) in a standalone commit to introduce the changes gradually, more granular == easier to review/revert/bisect